drivers/spi/spi_nxp_lpspi/spi_nxp_lpspi_common.c

```diff
@@ -101,17 +122,180 @@ static inline int lpspi_validate_xfer_args(const struct spi_config *spi_cfg)
 	return 0;
 }
 
-int spi_mcux_configure(const struct device *dev, const struct spi_config *spi_cfg)
+static uint8_t lpspi_calc_delay_scaler(uint32_t desired_delay_ns,
+				       uint32_t prescaled_clock,
+				       /* what the min_cycles parameter is about is
+					* that PCSSCK and SCKPSC are +1 cycles of the
+					* programmed value,
+					* while DBT is +2 cycles of the programmed value.
+					* Minimum value of min_cycles should be 1.
+					*/
+				       uint32_t min_cycles)
+{
+	/* the scaler value is how many cycles of the prescaled clock to delay for.
+	 * We most likely want to treat the requested delay as a minimum, since it
+	 * probably has to do with setup and hold times of some spi device.
+	 */
+	uint16_t max_cycles = 255 + min_cycles;
+	uint64_t delay_cycles;
+
+	delay_cycles = (uint64_t)prescaled_clock * desired_delay_ns;
+	delay_cycles = DIV_ROUND_UP(delay_cycles, NSEC_PER_SEC);
+	delay_cycles = CLAMP(delay_cycles, min_cycles, max_cycles);
+
+	return delay_cycles - min_cycles;
```

Comment:
Better safe than sorry?

```suggestion
	delay_cycles = CLAMP(delay_cycles, min_cycles, max_cycles);
	delay_cycles -= min_cycles;

	return MIN(delay_cycles, UINT8_MAX);
```

Reply:
`max_cycles` = 255 + `min_cycles`

delay_cycles gets clamped already to high of `max_cycles`, so assuming that is the maximum:

`max_cycles` - 255 = `min_cycles` -> 
`max_cycles` - `min_cycles` = 255
maximum value returned then would be 255

I won't fight too hard against making the change if you still think it's worthwhile but I think it's pretty obviously mathematically proveable that it won't be needed...

However, an argument could be made though that it could make the code above that more robust against someone changing it later erroneously. And it's negligible overhead to add that macro. What do you think

Reply:
I have an idea actually to satisfy your concern without adding extra code, I'll push it in a minute